src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/math/RobustBrentSolver.java

```diff
@@ -0,0 +1,439 @@
+        final double[] fSearchGrid = new double[xSearchGrid.length];
+        for (int k = 0; k < xSearchGrid.length; k++) {
+            fSearchGrid[k] = objFunc.value(xSearchGrid[k]);
+            evaluations++;
+        }
+
+        final List<Bracket> bracketsList = detectBrackets(xSearchGrid, fSearchGrid);
+        if (bracketsList.isEmpty()) {
+            throw new NoBracketingException(min, max, fSearchGrid[0], fSearchGrid[fSearchGrid.length-1]);
+        } else {
+            final List<Double> roots = new ArrayList<>(bracketsList.size());
+            for (final Bracket bracket : bracketsList) {
+                final BrentSolverWithInitialFuncEvals solver = new BrentSolverWithInitialFuncEvals(relativeAccuracy,
+                        absoluteAccuracy, functionValueAccuracy);
+                try {
+                    final double x0 = 0.5 * (bracket.xMin + bracket.xMax);
+                    final double x = solver.solve(maxEval, objFunc, bracket.xMin, bracket.xMax, x0,
+                            bracket.fMin, bracket.fMax, objFunc.value(x0));
+                    roots.add(x);
+                } catch (final TooManyEvaluationsException | NoBracketingException ex) {
+                    /* nothing to do */
+                }
+                evaluations += solver.getEvaluations();
+            }
+            if (roots.isEmpty()) {
+                throw new NoBracketingException(min, max, fSearchGrid[0], fSearchGrid[fSearchGrid.length-1]);
+            } else if (roots.size() == 1) {
+                return roots.get(0);
+            } else {
+                /* calculate merits */
+                if (meritFunc != null) {
+                    final List<Double> merits = roots.stream().mapToDouble(meritFunc::value).boxed().collect(Collectors.toList());
+                    evaluations += merits.size();
+                    double maxMerit = merits.get(0);
+                    int maxMeritIndex = 0;
+                    for (int rootIndex = 1; rootIndex < roots.size(); rootIndex++) {
+                        if (merits.get(rootIndex) > maxMerit) {
+                            maxMerit = merits.get(rootIndex);
+                            maxMeritIndex = rootIndex;
+                        }
+                    }
+                    return roots.get(maxMeritIndex);
+                } else {
+                    switch (meritPolicy) {
+                        case SMALLEST_ROOT:
+                            return Collections.min(roots);
+                        case LARGEST_ROOT:
+                            return Collections.max(roots);
+                        default:
+                            return Double.NaN;
+                    }
+                }
+            }
+        }
+    }
+
+    public int getEvaluations() {
+        return evaluations;
+    }
+
+    @VisibleForTesting
+    double[] createFractalSearchGrid(final double min, final double max, final int numBisections,
```

Comment:
Actually, is this method working as intended?  It seems to create a grid that is not symmetric in the interval.  For example, the result of your unit test is:
    
    0.0 4.8828125E-4 9.765625E-4 0.00146484375 0.001953125 0.0029296875 0.00390625 0.005859375 0.0078125 0.01171875 0.015625 0.0234375 0.03125 0.046875 0.0625 0.09375 0.125 0.1875 0.25 0.375 0.5 0.75 1.0